.github/workflows/update-sdk.yaml

```diff
@@ -0,0 +1,145 @@
+          sudo mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/debian/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+          echo \
+            "deb [signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+            $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+          sudo apt-get update
+          sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+
+      - uses: actions/checkout@v3
+        with:
+          path: scripts
+          fetch-depth: 0
+
+      - name: Set environment
+        shell: bash
+        run: |
+          if [ -n "${{ github.event.inputs.source_sdk_version }}" ] ; then
+              echo "SOURCE_SDK_VERSION=${{ github.event.inputs.source_sdk_version }}" >> $GITHUB_ENV
+          fi
+          if [ -n "${{ github.event.inputs.custom_sdk_version }}" ] ; then
+              echo "CUSTOM_SDK_VERSION=${{ github.event.inputs.custom_sdk_version }}" >> $GITHUB_ENV
+          fi
+
+      - name: Build an updated SDK container
+        shell: bash
+        run: |
+          exec 2>&1
+          set -x
+          set -euo pipefail
+
+          source ci-automation/ci_automation_common.sh
+          source sdk_container/.repo/manifests/version.txt
+
+          version="alpha-$FLATCAR_VERSION_ID"
+          sdk_version="${SOURCE_SDK_VERSION:-$FLATCAR_SDK_VERSION}"
+
+          sdk_name="flatcar-sdk-all"
+          docker_sdk_vernum="$(vernum_to_docker_image_version "${sdk_version}")"
+
+          docker_image_from_registry_or_buildcache "${sdk_name}" "${docker_sdk_vernum}"
+
+          sdk_image="$(docker_image_fullname "${sdk_name}" "${docker_sdk_vernum}")"
+
+          # Create version file
+          (
+            source sdk_lib/sdk_container_common.sh
+            create_versionfile "$sdk_version" "$version"
+          )
+
+          if [ -z "${CUSTOM_SDK_VERSION:-}" ] ; then
+              if [ -n "${{ github.event.issue.pull_request }}" ] ; then
+                target_version="${sdk_version}-github-PR-${{ github.event.issue.number }}-$(date '+%Y_%m_%d__%H_%M_%S')"
+              else
+                target_version="${sdk_version}-github-$(date '+%Y_%m_%d__%H_%M_%S')"
+              fi
+          else
+              target_version="${CUSTOM_SDK_VERSION}"
+          fi
+
+          echo "sdk_version=${target_version}" >> "$GITHUB_OUTPUT"
+
+          # This also updates sdk_container/.repo/manifests/version.txt with the new SDK version.
+          ./update_sdk_container_image "${target_version}"
+
+      - name: Upload the SDK container and binary packages to bincache
+        shell: bash
+        run: |
+          set -euo pipefail
+
+          source ci-automation/ci_automation_common.sh
+
+          mkdir -p ~/.ssh
+          trap 'rm -f ~/.ssh/bincache' EXIT
+          echo "${{ secrets.BINCACHESSH }}" > ~/.ssh/bincache
```

Comment:
There's some risk attached as GitHub Actions could be misconfigured to allow extraction when it runs without manual approval required. Another option would be to save this as regular CI artifact. What do you think?

Reply:
(Plus we don't need to think about extra garbage collection)

Reply:
I understand the concern and I realise it's a trade-off.
My concerns are that using github artifacts will make the overall process a lot slower - a lot of time is already spent on copying back and forth image build artifacts. Bincache access on the other hand is  significantly faster - often by a factor of 10 or higher.
Also, it's not unlikely that - even though officially Artifacts storage is unlimited for public repos - we'd eventually create friction with Github since we'd potentially produce hundreds of gigabytes of artifacts per week.

Reply:
Ok, maybe I don't know all plans, I thought that building an SDK is a rare even and while bigger than the regular images, it's still not that much of a difference.

Reply:
At the end of the day this is just one building block for fully automating merging of auto-generated PRs.
I think quite a few of the PRs that we're generating currently - kernel, rust, bulk Gentoo packages refreshes - would mandate an SDK update if we really want to test these fully automatedly.

But you're right with your concern, and one way around it would be to run the SDK container update as an optional step in the main `ci.yaml`. I'll think through it.

Reply:
Adding support for the SDK tar ball as GitHub Action artifact would be interesting security-wise but I think we can merge it as is and maybe try this out in a follow-up (I don't know how easy that would be).